### PR TITLE
Fix for the bug #602

### PR DESCRIPTION
**Pull request by animeshsingh**
Clears some verbiage for  https://github.com/kubeflow/kfserving/issues/602

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/603)

### CODE REVIEW

File: docs/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ KFServing currently requires `Knative Serving` for auto-scaling, canary rollout,
 * You can follow the instructions on [Set up a kubernetes cluster and install Knative Serving](https://knative.dev/docs/install/) or
 [Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`. Observability plug-ins are good to have for monitoring.
 
-* If you already have `Istio` (e.g. from a Kubeflow install) then simply skip the `Istio` steps. For Kubeflow install, you can install `Knative Serving` v0.8 via
+* If you already have `Istio` or `Knative` (e.g. from a Kubeflow install) then you don't need to install them explictily. From Kubeflow v0.7 onwards, KNative 0.8 and Istio 1.1.17 are installed by default as part of the Kubeflow installation. For prior versions of Kubeflow, you can install `Knative Serving` v0.8 via
```
Comment (yuzisun) [inline docs/DEVELOPER_GUIDE.md]: `kubeflow/manifests` may soon be changed to v0.10 though.

Comment (animeshsingh) [reply docs/DEVELOPER_GUIDE.md]: @yuzisun take a look - simplified the verbiage more, and removed any reference to manifests, as it was causing more confusion

Comment (yuzisun) [reply docs/DEVELOPER_GUIDE.md]: I think the istio version in 0.7 is 1.1.6 according to here https://github.com/kubeflow/manifests/blob/v0.7-branch/istio/istio-install/base/kustomization.yaml ?

Comment (animeshsingh) [reply docs/DEVELOPER_GUIDE.md]: @yuzisun corrected and rebased

Comment (yuzisun) [reply docs/DEVELOPER_GUIDE.md]: /1.1.16/1.1.6/

Comment (animeshsingh) [reply docs/DEVELOPER_GUIDE.md]: done

**Pull request merged**